src/main/java/core/basesyntax/MyLinkedList.java

```diff
@@ -3,45 +3,158 @@
 import java.util.List;
 
 public class MyLinkedList<T> implements MyLinkedListInterface<T> {
+    private int size = 0;
+    private Node<T> head;
+    private Node<T> tail;
+
+    private static class Node<T> {
+        private T element;
+        private Node<T> next;
+        private Node<T> prev;
+
+        Node(Node<T> prev, T element, Node<T> next) {
+            this.element = element;
+            this.prev = prev;
+            this.next = next;
+
+        }
+    }
+
     @Override
     public void add(T value) {
+        final Node<T> l = tail;
+        final Node<T> newNode = new Node<>(l, value, null);
+        tail = newNode;
+        if (l == null) {
+            head = newNode;
+        } else {
+            l.next = newNode;
+        }
+        size++;
     }
 
     @Override
     public void add(T value, int index) {
+        checkIndexForAdding(index);
+        if (index == size) {
+            add(value);
+        } else {
+            final Node<T> beforeNewNode = findNodeOnIndex(index).prev;
+            final Node<T> newNode = new Node<>(beforeNewNode,value, findNodeOnIndex(index));
+            findNodeOnIndex(index).prev = newNode;
```

Comment:
Can we use findNodeOnIndex(index) only ones here? findNodeOnIndex it's quite laborious operation, it should not be performed several times unnecessarily